### PR TITLE
*: optimize temptable.SessionSnapshotInterceptor

### LINKED ISSUE #37000: optimize `temptable.SessionSnapshotInterceptor` to reduce cpu cost
## Enhancement

kYePjVdg8o

`SessionSnapshotInterceptor` will call `session.InfoSchema` which will introduce some extra performance cost. We do not need to call `session.InfoSchema` because it infoschema is present in provider.

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #37000

Problem Summary:
kYePjVdg8o

`SessionSnapshotInterceptor` will call `session.InfoSchema` which will introduce some performance cost. We do not need to call `session.InfoSchema` because it infoschema is present in provider.

### What is changed and how it works?

Use provider's infoschema instead of `session.InfoSchema`

### Check List

Tests 

- [x] Unit test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by tiancaiamao — APPROVED** [review]

File: sessiontxn/internal/txn.go [hunk 1]
```diff
@@ -62,9 +61,11 @@ func CommitBeforeEnterNewTxn(ctx context.Context, sctx sessionctx.Context) error
 }
 
 // GetSnapshotWithTS returns a snapshot with ts.
-func GetSnapshotWithTS(s sessionctx.Context, ts uint64) kv.Snapshot {
+func GetSnapshotWithTS(s sessionctx.Context, ts uint64, interceptor kv.SnapshotInterceptor) kv.Snapshot {
 	snap := s.GetStore().GetSnapshot(kv.Version{Ver: ts})
-	snap.SetOption(kv.SnapInterceptor, temptable.SessionSnapshotInterceptor(s))
+	if interceptor != nil {
+		snap.SetOption(kv.SnapInterceptor, interceptor)
```
Comment (SpadeA-Tang) [inline sessiontxn/internal/txn.go:67]: The old implementation always set the SnapInterceptor. What if interceptor is nil?

Comment (lcwangchao) [reply sessiontxn/internal/txn.go:67]: To make the code strong. Actually, interceptor is not nil for the current implement.

**Review by SpadeA-Tang — APPROVED** [review]

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**